account_test.go

```diff
@@ -0,0 +1,118 @@
+package main
+
+import (
+	"testing"
+	"time"
+
+	vesting "github.com/cosmos/cosmos-sdk/x/auth/vesting/types"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	auth "github.com/cosmos/cosmos-sdk/x/auth/types"
+	bank "github.com/cosmos/cosmos-sdk/x/bank/types"
+
+	"github.com/cockroachdb/apd/v2"
+	"github.com/stretchr/testify/require"
+)
+
+func TestRegenToCoins(t *testing.T) {
+	one, _, err := apd.NewFromString("1")
+	require.NoError(t, err)
+
+	coins, err := RegenToCoins(one)
+	require.NoError(t, err)
+	require.Equal(t, "1000000uregen", coins.String())
+}
+
+func TestToCosmosAccount(t *testing.T) {
+	addr0 := sdk.AccAddress("abcdefg012345")
+	addr1 := sdk.AccAddress("012345abcdefg")
+	five, _, err := apd.NewFromString("5")
+	require.NoError(t, err)
+	ten, _, err := apd.NewFromString("10")
+	require.NoError(t, err)
+
+	time0, err := time.Parse(time.RFC3339, "2021-05-21T00:00:00Z")
+	require.NoError(t, err)
+
+	time1, err := time.Parse(time.RFC3339, "2021-06-21T00:00:00Z")
+	require.NoError(t, err)
+
+	tests := []struct {
+		name    string
+		acc     Account
+		want    auth.AccountI
+		want1   *bank.Balance
+		wantErr bool
+	}{
```

Comment:
Can you add a 3rd test case with 3-4 different periods? Currently we don't catch the error whereby period.Length is always duration from `StartDate` and it should actually be duration from last distribution's `dist.Time` value.

Reply:
Done